### PR TITLE
Add tutorial for LOJ-1164 Horrible Queries 

### PR DESCRIPTION
**Pull request by Evan-CSE**
closes: #471
Problem link: [Horrible Queries](https://lightoj.com/problem/horrible-queries)

Description: 
Wrote an English tutorial for the problem.

### CODE REVIEW

**Comment by Evan-CSE** [comment]
@rebornplusplus 
Hi! Can you check it now? Sorry for the inconvenience. It got messed up with my previous tutorial.

**Comment by Evan-CSE** [comment]
@rebornplusplus

File: 1164/en.md [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+                cout << query(1, 0, n - 1, left, right) <<endl;
+            }
+        }
+    }
+    return 0;
+}
+```
+## Resources:
+- [Segment Tree](https://cp-algorithms.com/data_structures/segment_tree.html)
+- [Lazy Propagation](https://www.topcoder.com/thrive/articles/range-operations-lazy-propagation)
```
Comment (rebornplusplus) [inline 1164/en.md:127]: Can we please just delete this block here and add the links to the items in the Prerequisites section, as the items remain same in both sections?

File: 1164/en.md [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+## Approach
+Its clear that we can use `cumulative sum` to answer the second type of query. However, updating a range of values using this approach would have a time complexity of **O(n)**, which could lead to a **"Time Limit Exceeded" (TLE)** error for larger inputs. If you are already familiar with **segment tree** then you can start with segment tree. But unlike segment tree we are required to **update range** which may lead to **TLE** To solve this problem we can use a variation of **segment tree** called **lazy propagation** 
+
+## Lazy Propagation at a glance
+Lazy propagation is an optimization technique used with segment trees. If you are not familiar with segment trees, it is recommended to learn about them before proceeding.
+
+With lazy propagation, instead of updating a range instantly, we only mark the root node of that range for future updates. We maintain an extra array to keep track of which nodes need to be updated. Here's an illustration:
+
+<img src="lazyPropagationExample.jpg">
```
Comment (rebornplusplus) [inline 1164/en.md:20]: Let's use the markdown format to insert an image.
```suggestion
![example image of lazy propagation](lazyPropagationExample.jpg)
```

File: 1164/en.md [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+Let's say we want to update the range [4:6]. The root node of this range is the 3rd node. Rather than updating all the nodes below [4:6] immediately, we simply mark the immediate children of the 3rd node with a value that should be added in the future. Our second array tracks this information for each node to determine if its child nodes should be updated or not. When we need to query or update the range again, we will update the specific node.
```
Comment (rebornplusplus) [inline 1164/en.md:22]: In the image above, can we please change the color of the nodes that will be marked during this lazy propagation

File: 1164/en.md [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+### Code
+
+```
+#include<bits/stdc++.h>
+
+using namespace std;
+
+#define ll long long
+#define endl "\n"
```
Comment (rebornplusplus) [inline 1164/en.md:34]: Let's please get rid of these custom definitions and values; use standard long forms instead.

**Review by rebornplusplus — CHANGES_REQUESTED** [review]
This overall looks okay, with a few minor comments here and there.

For future references, let's please also fill in the PR descriptions with appropriate info. For example, the default description set for the PR asks for an issue number, link to a problem and an optional changelog. They help the reviewers to get through to the problems and open issues quicker. Additionally, please wait for a few more days before subsequent pings.

File: 1164/en.md (hunk 3 above)
Comment (Evan-CSE) [reply 1164/en.md:22]: Actually, the image is collected from interenet. Based on this example I described the flow.

File: 1164/en.md (hunk 1 above)
Comment (Evan-CSE) [reply 1164/en.md:127]: done

File: 1164/en.md (hunk 2 above)
Comment (Evan-CSE) [reply 1164/en.md:20]: done

File: 1164/en.md (hunk 4 above)
Comment (Evan-CSE) [reply 1164/en.md:34]: done

**Comment by Evan-CSE** [comment]
@rebornplusplus Hi! Can you check it as per your convenience?

**Review by rebornplusplus — APPROVED** [review]
Looks good, thanks for the changes.

**Pull request merged**